generators/php.js

```diff
@@ -0,0 +1,207 @@
+
+/**
+ * Order of operation ENUMs.
+ * https://developer.mozilla.org/en/PHP/Reference/Operators/Operator_Precedence
+ */
+Blockly.PHP.ORDER_ATOMIC = 0;         // 0 "" ...
+Blockly.PHP.ORDER_CLONE = 1;          // clone
+Blockly.PHP.ORDER_NEW = 1;            // new
+Blockly.PHP.ORDER_MEMBER = 2;         // ()
+Blockly.PHP.ORDER_FUNCTION_CALL = 2;  // ()
+Blockly.PHP.ORDER_INCREMENT = 3;      // ++
+Blockly.PHP.ORDER_DECREMENT = 3;      // --
+Blockly.PHP.ORDER_LOGICAL_NOT = 4;    // !
+Blockly.PHP.ORDER_BITWISE_NOT = 4;    // ~
+Blockly.PHP.ORDER_UNARY_PLUS = 4;     // +
+Blockly.PHP.ORDER_UNARY_NEGATION = 4; // -
+Blockly.PHP.ORDER_TYPEOF = 4;         // typeof
+Blockly.PHP.ORDER_VOID = 4;           // void
+Blockly.PHP.ORDER_DELETE = 4;         // delete
+Blockly.PHP.ORDER_MULTIPLICATION = 5; // *
+Blockly.PHP.ORDER_DIVISION = 5;       // /
+Blockly.PHP.ORDER_MODULUS = 5;        // %
+Blockly.PHP.ORDER_ADDITION = 6;       // +
+Blockly.PHP.ORDER_SUBTRACTION = 6;    // -
+Blockly.PHP.ORDER_BITWISE_SHIFT = 7;  // << >> >>>
+Blockly.PHP.ORDER_RELATIONAL = 8;     // < <= > >=
+Blockly.PHP.ORDER_IN = 8;             // in
+Blockly.PHP.ORDER_INSTANCEOF = 8;     // instanceof
+Blockly.PHP.ORDER_EQUALITY = 9;       // == != === !==
+Blockly.PHP.ORDER_BITWISE_AND = 10;   // &
+Blockly.PHP.ORDER_BITWISE_XOR = 11;   // ^
+Blockly.PHP.ORDER_BITWISE_OR = 12;    // |
+Blockly.PHP.ORDER_CONDITIONAL = 13;   // ?:
+Blockly.PHP.ORDER_ASSIGNMENT = 14;    // = += -= *= /= %= <<= >>= ...
+Blockly.PHP.ORDER_LOGICAL_AND = 15;   // &&
+Blockly.PHP.ORDER_LOGICAL_OR = 16;    // ||
+Blockly.PHP.ORDER_COMMA = 17;         // ,
+Blockly.PHP.ORDER_NONE = 99;          // (...)
+
+/**
+ * Initialise the database of variable names.
+ * @param {!Blockly.Workspace} workspace Workspace to generate code from.
+ */
+Blockly.PHP.init = function(workspace) {
+  // Create a dictionary of definitions to be printed before the code.
+  Blockly.PHP.definitions_ = Object.create(null);
+  // Create a dictionary mapping desired function names in definitions_
+  // to actual function names (to avoid collisions with user functions).
+  Blockly.PHP.functionNames_ = Object.create(null);
+
+  if (!Blockly.PHP.variableDB_) {
+    Blockly.PHP.variableDB_ =
+        new Blockly.Names(Blockly.PHP.RESERVED_WORDS_);
+  } else {
+    Blockly.PHP.variableDB_.reset();
+  }
+};
+
+Blockly.PHP.getDistinctName = function(name, type) {
+  var safeName = this.variableDB_.safeName_(name);
+  var i = '';
+  while ((type == Blockly.Procedures.NAME_TYPE && (safeName + i) in this.variableDB_.reservedDict_)
+  || this.variableDB_.dbReverse_[safeName + i]) {
+    // Collision with existing name.  Create a unique name.
+    i = i ? i + 1 : 2;
+  }
+  safeName += i;
+  this.variableDB_.dbReverse_[safeName] = true;
+  if (type === Blockly.Procedures.NAME_TYPE) return safeName;
```

Comment:
Suggest reversing this:
if (type == Blockly.Variables.NAME_TYPE) {
  safeName =  '$' + safeName;
}
return safeName;